### PR TITLE
Add image pushing for CAPA and CAPG

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

/assign @detiber @Katharine

### CODE REVIEW

**Comment by vincepri** [comment]
CAPG: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/146
CAPA: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1123

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml [hunk 1]
```diff
@@ -30,3 +30,61 @@ postsubmits:
           - name: creds
             secret:
               secretName: deployer-service-account
+    - name: post-cluster-api-provider-aws-push-images
```
Comment (Katharine) [inline config/jobs/image-pushing/k8s-staging-cluster-api.yaml]: This looks like it applies to a different repo, so should be in its own block:

```yaml
  kubernetes-sigs/cluster-api-provider-aws:
    - name: post-cluster-api-provider-aws-push-images
      # ...
```

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml [hunk 2]
```diff
@@ -30,3 +30,61 @@ postsubmits:
...
+              - name: GOOGLE_APPLICATION_CREDENTIALS
+                value: /creds/service-account.json
+            volumeMounts:
+              - name: creds
+                mountPath: /creds
+        volumes:
+          - name: creds
+            secret:
+              secretName: deployer-service-account
+    - name: post-cluster-api-provider-gcp-push-images
```
Comment (Katharine) [inline config/jobs/image-pushing/k8s-staging-cluster-api.yaml]: Same as above:

```yaml
  kubernetes-sigs/cluster-api-provider-gcp:
    - name: post-cluster-api-provider-gcp-push-images
      # ...
```

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml [hunk 3]
```diff
@@ -30,3 +30,61 @@ postsubmits:
...
+        - ^master$
+      spec:
+        containers:
+          - image: gcr.io/k8s-testimages/image-builder:v20190906-d5d7ce3
+            command:
+              - /run.sh
+            args:
+              # this is the project GCB will run in, which is the same as the GCR images are pushed to.
+              - --project=k8s-staging-cluster-api
+              - --scratch-bucket=gs://k8s-staging-cluster-api-gcp-gcb
```
Comment (Katharine) [inline config/jobs/image-pushing/k8s-staging-cluster-api.yaml]: This should just be `<project>-gcb`. It's fine for multiple jobs to use the same bucket, if they're going into the same GCR.

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml [hunk 4]
```diff
@@ -30,3 +30,61 @@ postsubmits:
...
+        - ^master$
+      spec:
+        containers:
+          - image: gcr.io/k8s-testimages/image-builder:v20190906-d5d7ce3
+            command:
+              - /run.sh
+            args:
+              # this is the project GCB will run in, which is the same as the GCR images are pushed to.
+              - --project=k8s-staging-cluster-api
+              - --scratch-bucket=gs://k8s-staging-cluster-api-aws-gcb
```
Comment (Katharine) [inline config/jobs/image-pushing/k8s-staging-cluster-api.yaml]: This should just be `<project>-gcb`. It's fine for multiple jobs to use the same bucket, if they're going into the same GCR.

Comment (vincepri) [reply config/jobs/image-pushing/k8s-staging-cluster-api.yaml]: Definitely a mistake, good catch!

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml (hunk 3 above)
Comment (vincepri) [reply config/jobs/image-pushing/k8s-staging-cluster-api.yaml]: Same as above

**Comment by vincepri** [comment]
Related PRs have been merged ! 

/hold cancel

**Pull request merged**